lib/adyen/rest/request.rb

```diff
@@ -29,17 +29,19 @@ module REST
     # @see Adyen::REST::Client
     # @see Adyen::REST::Response
     class Request
-      attr_reader :prefix, :form_data, :required_attributes
+      attr_reader :prefix, :form_data, :required_attributes, :resource
       attr_accessor :response_class, :response_options
 
       def initialize(action, attributes, options = {})
-        @prefix = options[:prefix]
-        @form_data = generate_form_data(action, attributes)
+        raise Adyen::REST::RequiredParameterMissing, "actions is required!" if action.nil?
```

Comment:
This is not really needed, since a nil action will immediately fail on line 39. And anyway, you can't catch all potentially wrong values of action here.